OrbitLinuxTracing/PerfEventOpen.h

```diff
@@ -43,21 +45,32 @@ inline void perf_event_reset_and_enable(int file_descriptor) {
 inline void perf_event_disable(int file_descriptor) {
   int ret = ioctl(file_descriptor, PERF_EVENT_IOC_DISABLE, 0);
   if (ret != 0) {
-    ERROR("PERF_EVENT_IOC_DISABLE: %d", ret);
+    ERROR("PERF_EVENT_IOC_DISABLE: %s", SafeStrerror(errno));
   }
 }
 
 inline void perf_event_redirect(int from_fd, int to_fd) {
   int ret = ioctl(from_fd, PERF_EVENT_IOC_SET_OUTPUT, to_fd);
   if (ret != 0) {
-    ERROR("PERF_EVENT_IOC_SET_OUTPUT: %d\n", ret);
+    ERROR("PERF_EVENT_IOC_SET_OUTPUT: %s", SafeStrerror(errno));
   }
 }
 
-// This must be in sync with struct perf_event_sample_id_tid_time_cpu in
-// PerfEventRecords.h.
-static constexpr uint64_t SAMPLE_TYPE_TID_TIME_CPU =
-    PERF_SAMPLE_TID | PERF_SAMPLE_TIME | PERF_SAMPLE_CPU;
+inline uint64_t perf_event_get_id(int file_descriptor) {
+  uint64_t id;
+  int ret = ioctl(file_descriptor, PERF_EVENT_IOC_ID, &id);
+  if (ret != 0) {
+    ERROR("PERF_EVENT_IOC_ID: %s", SafeStrerror(errno));
+    return 0;
+  }
+  return id;
+}
+
+// This must be in sync with struct perf_event_sample_id_tid_time_streamid_cpu
```

Comment:
Seeing this, I'm not sure it's good practice to keep all options in the name of the type. This requires renaming this struct everywhere anytime you add an option here.

Reply:
As changing this struct affects all the structs in PerfEventRecords and has to be reflected in all the function in PerfEventOpen (through SAMPLE_TYPE_TID_TIME_STREAMID_CPU), the reasoning is that it is good to be very clear on what fields it contains (although the name is long and ugly) and that it has to be in sync with SAMPLE_TYPE_TID_TIME_STREAMID_CPU. Adding fields and hence renaming is very rare and shouldn't be an issue.